Automattic/wp-calypso
# Sites page: Fix nested dropdown clipped by the viewport

Closes https://github.com/Automattic/wp-calypso/issues/81590

## Proposed Changes

This PR fixes the nested dropdown from the `Hosting Configuration` menu item in the ellipsis menu on `https://wordpress.com/sites` that was getting clipped by the viewport.

**Before**

<img width="1243" alt="Screenshot 2023-09-12 at 11 02 53 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/8aa25c96-5e5c-41ab-956f-4de849e215c3">
   
**After**

<img width="1239" alt="Screenshot 2023-09-12 at 11 03 24 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/f78152dd-8582-40dc-93c5-e84af50617ed">

## Testing Instructions

**Testing the grid view**

* Navigate to `https://wordpress.com/sites` (you can use Calypso Live link)
* Select the grid view:

<img width="1210" alt="Screenshot 2023-09-12 at 11 07 45 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/7ea6c53e-2ac2-4d38-9e64-fa467cff4b36">

* Click on the ellipsis menu for a site that is located on the right side (make sure you don't use a P2 site as it does not have `Hosting Configuration` menu item)
* Select `Hosting Configuration` menu item
* Confirm that the nested popover does not get clipped on the right and flips to the left
* Check the popover for the other sites in the grid (the one on the left, the one in the middle)

**Testing the list view**

* Switch to the list view:

<img width="1229" alt="Screenshot 2023-09-12 at 11 10 16 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/8a88c33e-2605-4581-b857-e69675152aa7">

* Navigate to the ellipsis menu
* Select `Hosting Configuration` menu item
* Confirm that the nested popover does not get clipped by the viewport and flips to the left 

----

**Issue with styles**:

While fixing the viewport, I noticed one more regression with the nested submenu popover - the CSS cascade was broken and the styles were not applying. This is what the result of the regression looks like:

<img width="637" alt="Screenshot 2023-09-13 at 11 02 12 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/d8152988-9b11-41b1-b42d-e10c701b4ebe">

After some research, I found that the issue is coming from https://github.com/WordPress/gutenberg/pull/53982 and https://github.com/WordPress/gutenberg/pull/53889: recently the `Popover.Slot` was made optional which made the popover to be rendered at the end of the body instead of being rendered inline:

![Screenshot 2023-09-12 at 4 36 37 PM](https://github.com/Automattic/wp-calypso/assets/25575134/6dcb373e-fab3-4a40-ba6c-899748bc1b9e)

As a solution, I added an optional `inline` prop to make the popover render inline again and inherit the styles. This is what it looks like after the fix:

<img width="1528" alt="Screenshot 2023-09-13 at 11 06 59 AM" src="https://github.com/Automattic/wp-calypso/assets/25575134/a36c4556-8e36-4642-9390-90f65d95e2c9">

**To test**:

* Navigate to `https://wordpress.com/sites` (you can use Calypso Live link)
*  Click on the ellipsis menu for a site that is located on the right side (make sure you don't use a P2 site as it does not have `Hosting Configuration` menu item)
* Select `Hosting Configuration` menu item
* Confirm that the styles of the popover inherit the styles of the parent (the styles should look as on the screenshot above)

**Note**: I tried a few solutions for this and `inline` prop seemed to be the least messy of all but happy to reconsider if there are other preferences or ideas. 

## Pre-merge Checklist
